### PR TITLE
3.1.7 Add docs for Configure::readOrFail()

### PR DESCRIPTION
**Pull request by ionas**
Ref: https://github.com/cakephp/cakephp/pull/7935

### CODE REVIEW

File: en/development/configuration.rst [hunk 1]
```diff
@@ -282,6 +282,19 @@ data back::
 
 If $key is left null, all values in Configure will be returned.
 
+.. php:staticmethod:: readOrFail($key)
+
+Reads configuration data just php:staticmethod:: read but expects to find a key/value pair.
```
Comment (markstory) [inline en/development/configuration.rst]: Link should be ``:php:method:`Cake\\Core\\Configure::read```

File: en/development/configuration.rst [hunk 2]
```diff
@@ -282,6 +282,19 @@ data back::
...
+In case the requested pair does not exist, a RuntimeException will be thrown.::
```
Comment (markstory) [inline en/development/configuration.rst]: No `.` before the `::`.

File: en/development/configuration.rst (hunk 1 above)
Comment (ionas) [inline en/development/configuration.rst]: I don’t understand a few lines up it is ```php:staticmethod:: read($key = null)``` ... so I added .. ```php:staticmethod:: readOrFail($key = null)```.

File: en/development/configuration.rst [hunk 3]
```diff
@@ -282,6 +282,21 @@ data back::
 
 If $key is left null, all values in Configure will be returned.
 
+.. versionadded:: 3.1.7
+
+.. php:staticmethod:: readOrFail($key)
+
+Reads configuration data just like :php:method:`Cake\\Core\\Configure::read` but expects to find a key/value pair.
```
Comment (antograssiot) [inline en/development/configuration.rst]: it's ``:php:meth`` not ``:php:method:`` so it cause the build to fail

File: en/development/configuration.rst [hunk 4]
```diff
@@ -282,6 +282,21 @@ data back::
...
+In case the requested pair does not exist, a :php:class:`RuntimeException` will be thrown.
+
```
Comment (antograssiot) [inline en/development/configuration.rst]: you need to a ``.. code-block:: php``here so the code bellow can be correctly formated

File: en/development/configuration.rst (hunk 3 above)
Comment (ionas) [inline en/development/configuration.rst]: Link not working. But i fixed it.

File: en/development/configuration.rst (hunk 4 above)
Comment (ionas) [inline en/development/configuration.rst]: I fixed it back to how it is a few lines above. Someone told me to remove `::` before. Working like this?

File: en/development/configuration.rst (hunk 1 above)
Comment (markstory) [inline en/development/configuration.rst]: The `php:staticmethod:: read` defines the method link target and formats the method heading. The code you originally had looked like a 'link', but id now a method definition which is fine.

**Pull request merged**